PinpointKit/PinpointKitTests/SystemLogCollectorTests.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  SystemLogCollectorTests.swift
+//  PinpointKit
+//
+//  Created by Andrew Harrison on 8/5/16.
+//  Copyright © 2016 Lickability. All rights reserved.
+//
+
+import XCTest
+@testable import PinpointKit
+
+class SystemLogCollectorTests: XCTestCase {
+    
+    func testLogCollectorCollectsLogs() {
+        let testString = "TestLog"
+        let sut = SystemLogCollector(loggingType: .Testing)
+        
+        NSLog(testString)
+        NSLog(testString)
+        NSLog(testString)
+        
+        let expectation = defaultExpectation()
+        
+        let delayTime = dispatch_time(DISPATCH_TIME_NOW, Int64(0.1 * Double(NSEC_PER_SEC)))
+        dispatch_after(delayTime, dispatch_get_main_queue()) {
+            let logs = sut.retrieveLogs()
+            
+            guard let firstLog = logs.first else { return XCTFail("There should be at least 1 log.") }
+            
+            XCTAssertEqual(logs.count, 3)
+            XCTAssertTrue(firstLog.containsString(testString))
+            expectation.fulfill()
+        }
+        
+        waitForExpectationsWithTimeout(5, handler: nil)
+        
+    }
+    
+    func testLogCollectorCollecsLogInOrder() {
+        let testString1 = "First"
+        let testString2 = "Second"
+        let testString3 = "Third"
+        
+        let sut = SystemLogCollector(loggingType: .Testing)
+        
+        NSLog(testString1)
+        NSLog(testString2)
+        NSLog(testString3)
+        
+        let expectation = defaultExpectation()
+        
+        let delayTime = dispatch_time(DISPATCH_TIME_NOW, Int64(0.1 * Double(NSEC_PER_SEC)))
+        dispatch_after(delayTime, dispatch_get_main_queue()) {
+            let logs = sut.retrieveLogs()
+            
+            guard let firstLog = logs[optional: 0] else { return XCTFail("There should be a first log.") }
```

Comment:
The optional subscript is pretty cool, but it might be more accessible to check and assert that the count is what you expect and use the default subscript.